pkg/controllers/deprovisioning/controller.go

```diff
@@ -411,57 +415,61 @@ func (c *Controller) waitForDeletion(ctx context.Context, node *v1.Node) {
+	if len(nodeNames) != len(action.replacementNodes) {
+		// shouldn't ever occur since a partially failed LaunchNodes should return an error
+		return fmt.Errorf("expected %d node names, got %d", len(action.replacementNodes), len(nodeNames))
 	}
-
-	metrics.NodesCreatedCounter.WithLabelValues(metrics.DeprovisioningReason).Inc()
-
-	// We have the new node created at the API server so mark the old node for deletion
-	c.cluster.MarkForDeletion(oldNode.Name)
-
-	var k8Node v1.Node
-	// Wait for the node to be ready
-	var once sync.Once
-	if err := retry.Do(func() error {
-		if err := c.kubeClient.Get(ctx, client.ObjectKey{Name: nodeNames[0]}, &k8Node); err != nil {
-			return fmt.Errorf("getting node, %w", err)
-		}
-		once.Do(func() {
-			c.recorder.Publish(deprovisioningevents.LaunchingNode(&k8Node, action.String()))
-		})
-
-		if _, ok := k8Node.Labels[v1alpha5.LabelNodeInitialized]; !ok {
-			// make the user aware of why deprovisioning is paused
-			c.recorder.Publish(deprovisioningevents.WaitingOnReadiness(&k8Node))
-			return fmt.Errorf("node is not initialized")
+	metrics.NodesCreatedCounter.WithLabelValues(metrics.DeprovisioningReason).Add(float64(len(nodeNames)))
+
+	// We have the new nodes created at the API server so mark the old nodes for deletion
+	c.cluster.MarkForDeletion(nodeNamesToRemove...)
+	// Wait for nodes to be ready
+	// TODO @njtran: Allow to bypass this check for certain deprovisioners
+	errs := make([]error, len(nodeNames))
+	workqueue.ParallelizeUntil(ctx, len(nodeNames), len(nodeNames), func(i int) {
+		var k8Node v1.Node
+		// Wait for the node to be ready
+		var once sync.Once
+		if err := retry.Do(func() error {
+			if err := c.kubeClient.Get(ctx, client.ObjectKey{Name: nodeNames[i]}, &k8Node); err != nil {
+				return fmt.Errorf("getting node, %w", err)
+			}
+			once.Do(func() {
+				c.recorder.Publish(deprovisioningevents.LaunchingNode(&k8Node, action.String()))
+			})
+
+			if _, ok := k8Node.Labels[v1alpha5.LabelNodeInitialized]; !ok {
+				// make the user aware of why deprovisioning is paused
+				c.recorder.Publish(deprovisioningevents.WaitingOnReadiness(&k8Node))
+				return fmt.Errorf("node is not initialized")
+			}
+			return nil
+		}, waitRetryOptions...); err != nil {
+			// nodes never become ready, so uncordon the nodes we were trying to delete and report the error
+			errs[i] = err
 		}
-		return nil
-	}, waitRetryOptions...); err != nil {
-		// node never become ready, so uncordon the node we were trying to delete and report the error
-		c.cluster.UnmarkForDeletion(oldNode.Name)
-		return multierr.Combine(c.setNodeUnschedulable(ctx, oldNode.Name, false),
-			fmt.Errorf("timed out checking node readiness, %w", err))
+	})
+	multiErr := multierr.Combine(errs...)
+	if multiErr != nil {
+		c.cluster.UnmarkForDeletion(nodeNamesToRemove...)
+		return multierr.Combine(c.setNodesUnschedulable(ctx, false, nodeNamesToRemove...),
+			fmt.Errorf("timed out checking node readiness. %w", multiErr))
```

Comment:
```suggestion
			fmt.Errorf("timed out checking node readiness, %w", multiErr))
```

nit: typically wrapped errors are prepended with a comma

Reply:
Yeah, typo. Good catch, new keyboard.